.github/workflows/maven-verify.yml

```diff
@@ -199,7 +212,8 @@ jobs:
     name: ${{ matrix.os }} jdk-${{ matrix.jdk }}-${{ matrix.distribution }}
     timeout-minutes: ${{ inputs.timeout-minutes }}
     runs-on: ${{ matrix.os }}
-
+    if: always() && ( !inputs.ff-run || needs.fail-fast-build.result == 'success' )
```

Comment:
It seems this `if: always()` here has the unfortunate side effect that it is impossible to cancel the workflow, see https://github.com/orgs/community/discussions/26303. That is quite problematic because once the matrix starts, you have potentially dozens of long running uncancelable jobs.
As suggested in that discussion, maybe `!cancelled()` could be an alternative to `always()` here?

What was the intention for `if: always()`? Note that there is already `strategy.fail-fast` being set below.

Sorry for commenting on this old PR, but since this introduced it originally, I hope that is ok.